server/tests/test_submission_statuses.py

```diff
@@ -84,102 +87,151 @@ def create_data(abc, xyz=NO_DATA_TRIPLE, ijk=NO_DATA_TRIPLE):
+            xyz_dim.id: SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
+            ijk_dim.id: SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
+        },
+        "program_global_status": SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
+    }
+
+
+def test_both_submitted():
+    ind, period, abc_dim, xyz_dim, ijk_dim = create_data(
+        abc=["both_submitted"] * 3,
+        xyz=["both_submitted"] * 3,
+        ijk=["both_submitted"] * 3,
     )
     statuses = get_submission_statuses(ind, period)
     assert statuses == {
-        "statuses_by_dimension_type_id": {
+        "hso_statuses_by_dimension_type_id": {
             abc_dim.id: SUBMISSION_STATUSES.SUBMITTED,
             xyz_dim.id: SUBMISSION_STATUSES.SUBMITTED,
+            ijk_dim.id: SUBMISSION_STATUSES.SUBMITTED,
+        },
+        "hso_global_status": SUBMISSION_STATUSES.SUBMITTED,
+        "program_statuses_by_dimension_type_id": {
+            abc_dim.id: SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
+            xyz_dim.id: SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
             ijk_dim.id: SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
         },
-        "global_status": SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
+        "program_global_status": SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
     }
 
 
-def test_fully_submitted_with_edit():
+def test_both_submitted_with_edit():
     ind, period, abc_dim, xyz_dim, ijk_dim = create_data(
-        abc=[SUBMISSION_STATUSES.SUBMITTED] * 3,
-        xyz=[SUBMISSION_STATUSES.SUBMITTED] * 3,
+        abc=["both_submitted"] * 3,
+        xyz=["both_submitted"] * 3,
         ijk=[
-            SUBMISSION_STATUSES.SUBMITTED,
-            SUBMISSION_STATUSES.MODIFIED_SINCE_LAST_SUBMISSION,
-            SUBMISSION_STATUSES.SUBMITTED,
+            "both_submitted",
+            "both_modified",
+            "both_submitted",
         ],
     )
     statuses = get_submission_statuses(ind, period)
     assert statuses == {
-        "statuses_by_dimension_type_id": {
+        "hso_statuses_by_dimension_type_id": {
             abc_dim.id: SUBMISSION_STATUSES.SUBMITTED,
             xyz_dim.id: SUBMISSION_STATUSES.SUBMITTED,
             ijk_dim.id: SUBMISSION_STATUSES.MODIFIED_SINCE_LAST_SUBMISSION,
         },
-        "global_status": SUBMISSION_STATUSES.MODIFIED_SINCE_LAST_SUBMISSION,
+        "hso_global_status": SUBMISSION_STATUSES.MODIFIED_SINCE_LAST_SUBMISSION,
+        "program_statuses_by_dimension_type_id": {
+            abc_dim.id: SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
+            xyz_dim.id: SUBMISSION_STATUSES.PROGRAM_SUBMITTED,
+            ijk_dim.id: SUBMISSION_STATUSES.MODIFIED_SINCE_LAST_SUBMISSION,
+        },
+        "program_global_status": SUBMISSION_STATUSES.MODIFIED_SINCE_LAST_SUBMISSION,
     }
 
 
-def test_modified_takes_precedence_over_program_submitted():
+def test_hso_submitted_with_edit():
     ind, period, abc_dim, xyz_dim, ijk_dim = create_data(
-        abc=[SUBMISSION_STATUSES.SUBMITTED] * 3,
```

Comment:
Is there a reason you replaced the constants with string literals? Is it that the options have changed?

Reply:
Yes, 
Its because options have changed and program and hso statuses are independent now

For example,
SUBMISSION_STATUSES.PROGRAM_SUBMITTED used to mean that the data has been submitted by program but not by HSO
but now it means data has been submitted by program but it gives no insight into hso submission status.